internal/trafficcontrol/tc.go

```diff
@@ -201,3 +212,33 @@ func IsSriovPF(ifName string) bool {
 	}
 	return true
 }
+
+func GetSriovPFInfo(ifName string) (*PFInfo, error) {
+	if !IsSriovPF(ifName) {
+		return nil, fmt.Errorf("interface %s is not a sriov pf type", ifName)
+	}
+	pfLink, err := netlink.LinkByName(ifName)
+	if err != nil {
+		return nil, err
+	}
+	pf := &PFInfo{Name: ifName, Authenticated: false, VFs: map[int]*VFInfo{}}
+	for _, vf := range pfLink.Attrs().Vfs {
+		pf.VFs[vf.ID] = &VFInfo{Index: vf.ID, Vlan: vf.Vlan}
+	}
+	return pf, nil
+}
+
+func ConfigureVF(logger log.Logger, pfInfo *PFInfo, vfInfo *VFInfo) error {
+	pfLink, err := netlink.LinkByName(pfInfo.Name)
+	if err != nil {
+		return err
+	}
+	var vlan int
+	if !pfInfo.Authenticated {
+		// Use reserved vlan value 4095 when PF is in unauthenticated state.
+		vlan = 4095
+	} else {
+		vlan = vfInfo.Vlan
+	}
+	return netlink.LinkSetVfVlan(pfLink, vfInfo.Index, vlan)
```

Comment:
I think we now need to also call `netlink.LinkSetVfState()` to either `VF_LINK_STATE_AUTO` or `VF_LINK_STATE_DISABLE` depending on the auth state?

Reply:
oh! yes, forgot about link state. changed it now.